pyroomacoustics/doa/doa.py

```diff
@@ -383,6 +383,11 @@ def polar_plt_dirac(self, azimuth_ref=None, alpha_ref=None, save_fig=False,
             'dirty image' in the case of FRI.
         """
 
+        if not matplotlib_available:
+            import warnings
+            warnings.warn('Matplotlib is required for plotting')
+            return
```

Comment:
Could be alright, but wondering: why do you raise a warning and return _instead_ of just raising an error?

Reply:
My thinking is that we don't want the script to fail just because matplotlib is missing. For example when running simulation on a server. This is not a requirement after all :) So raise a warning and silently return.